evm/db/shard.py

```diff
@@ -35,75 +36,102 @@ def __init__(self, db):
         self.db = db
 
     #
-    # Collation and header API
+    # Collation Getters by Hash
     #
-    def get_header(self, shard_id: int, period: int) -> CollationHeader:
-        key = make_collation_header_lookup_key(shard_id, period)
+    def get_header_by_hash(self, collation_hash: bytes) -> CollationHeader:
         try:
-            header = self.db.get(key)
+            header = self.db.get(collation_hash)
         except KeyError:
-            raise CollationHeaderNotFound("No header for shard {} and period {} found".format(
-                shard_id,
-                period,
-            ))
+            raise CollationHeaderNotFound("No header with hash {} found".format(collation_hash))
         return rlp.decode(header, sedes=CollationHeader)
 
-    def get_body(self, shard_id: int, period: int) -> bytes:
-        key = make_collation_body_lookup_key(shard_id, period)
+    def get_body_by_chunk_root(self, chunk_root: bytes) -> bytes:
         try:
-            body = self.db.get(key)
+            body = self.db.get(chunk_root)
         except KeyError:
-            raise CollationBodyNotFound("No body for shard {} and period {} found".format(
-                shard_id,
-                period,
-            ))
+            raise CollationBodyNotFound("No body with chunk root {} found".format(chunk_root))
         return body
 
-    def get_collation(self, shard_id: int, period: int) -> Collation:
-        header = self.get_header(shard_id, period)
-        body = self.get_body(shard_id, period)
+    def get_collation_by_hash(self, collation_hash: bytes) -> Collation:
+        header = self.get_header_by_hash(collation_hash)
+        body = self.get_body_by_chunk_root(header.chunk_root)
         return Collation(header, body)
 
+    #
+    # Canonical Collations
+    #
+    def mark_canonical(self, header: CollationHeader) -> None:
```

Comment:
maybe use `set_canonical_collation` to stay more inline with the `get/set` naming convention.